src/client/Microsoft.Identity.Client/AppConfig/AuthorityInfo.cs

```diff
@@ -91,6 +116,10 @@ internal class AuthorityInfo
         public string UserRealmUriPrefix { get; }
         public bool ValidateAuthority { get; }
 
+        internal bool IsInstanceDiscoverySupported => (AuthorityType == AuthorityType.Aad);
+
+        internal bool IsServicePricipalObOSupported => (AuthorityType == AuthorityType.Aad || AuthorityType == AuthorityType.Dsts);
```

Comment:
I was thinking more about this flag and I think what we really want is "SupportsClientInfo". Because we always ask for client_info, which returns an identifier for the user (UPN is not good enough, AAD returns "oid.tid"). 

ADFS does not return client_info. 
AAD, B2C and DSTS return client_info.

Reply:
SupportsClientInfo would be more appropriate. I suggest start with Is i.e. IsClientInfoSupported?